src/rgw/rgw_gc.cc

```diff
@@ -38,15 +38,20 @@ void RGWGC::initialize(CephContext *_cct, RGWRados *_store) {
     char buf[32];
     snprintf(buf, 32, ".%d", i);
     obj_names[i].append(buf);
-  }
 
-  // Create GC queues
-  for (int i = 0; i < max_objs; i++) {
-    ldpp_dout(this, 20) << "RGWGC::initialize initing gc queue with name = " << obj_names[i] << dendl;
+    auto it = transitioned_objects_cache.begin() + i;
+    transitioned_objects_cache.insert(it, false);
+
+    //version = 0 -> not ready for transition
+    //version = 1 -> marked ready for transition
     librados::ObjectWriteOperation op;
-    op.create(true);
-    uint64_t queue_size = 1048576, num_urgent_data_entries = 50;
+    obj_version objv;
+    cls_version_check(op, objv, VER_COND_EQ);
+    op.create(false);
```

Comment:
@cbodley : Do we need op.create(), since rados automatically creates an object for every write op?

Reply:
it isn't required, but i like being explicit about it - i'd just put it before the cls_version_check()

Reply:
Yeah realised that after submitting the PR (the order), will correct it.